spring-boot-actuator/src/main/java/org/springframework/boot/actuate/health/RedisHealthIndicator.java

```diff
@@ -69,50 +64,14 @@ protected void doHealthCheck(Health.Builder builder) throws Exception {
 		}
 	}
 
-	private void redisClusterInfo(Health.Builder builder,
-			RedisClusterConnection clusterConnection) {
-		Iterable<RedisClusterNode> clusterNodes = clusterConnection.clusterGetNodes();
-		List<RedisNodeInfo> nodes = new ArrayList<RedisNodeInfo>();
-		for (RedisClusterNode clusterNode : clusterNodes) {
-			Properties info = clusterConnection.info(clusterNode);
-			nodes.add(new RedisNodeInfo(clusterNode.toString(),
-					info.getProperty(REDIS_VERSION), info.getProperty(ROLE)));
-		}
-		defaultRedisInfo(builder, NODES, nodes);
+	private void redisClusterInfo(Health.Builder builder, ClusterInfo clusterInfo) {
+		defaultRedisInfo(builder, "cluster_size", clusterInfo.getClusterSize());
+		defaultRedisInfo(builder, "up_nodes", clusterInfo.getSlotsOk());
```

Comment:
Slots aren't nodes in Redis Cluster. Slots are logical partitions, so `slots_up` and `slots_fail` would fit better.